### PR TITLE
Added character with controller and animations

### PR DESCRIPTION
**Pull request by kdreese**

### CODE REVIEW

**Comment by kdreese** [comment]
Dash ability is currently broken and all the numbers need to be tweaked, but this is a good start to the character controller

File: src/player.gd [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+@export var unlocked_dash := false
+var is_dashing := false
+var dash_time := 0.0
+
+
+func _physics_process(delta: float) -> void:
+	var acceleration
+	var deceleration
+	var turn_speed
+	var max_speed_change
```
Comment (admoore0) [inline src/player.gd]: Declare types!

File: src/player.gd [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+			else:
+				max_speed_change = acceleration * delta
+		else:
+				max_speed_change = deceleration * delta
+		velocity.x = move_toward(velocity.x, direction * SPEED, max_speed_change)
+
+	move_and_slide()
+
+
+func Jump() -> void:
```
Comment (admoore0) [inline src/player.gd]: what is this, C#?

**Review by thearst3rd — DISMISSED** [review]

File: src/player.gd [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+		Jump()
+		has_used_grounded_jump = true
+	if Input.is_action_just_pressed("jump") and can_air_jump:
+		Jump()
+		has_used_aerial_jump = true
+	if Input.is_action_just_pressed("jump") and not (can_ground_jump or can_air_jump):
+		stored_jump_attempt = true
+	
+	# Handle coyote frames
+	if not is_on_floor() and time_since_was_on_floor <= COYOTE_TIME_LIMIT:
```
Comment (admoore0) [inline src/player.gd]: Turn into its own function `was_on_floor()`

**Review by admoore0 — CHANGES_REQUESTED** [review]

**Review by admoore0 — APPROVED** [review]

**Review by thearst3rd — APPROVED** [review]

**Pull request merged**